app/lib/screens/HomeScreens/ChatScreen.dart

```diff
@@ -0,0 +1,435 @@
+              builder: (
+                BuildContext context,
+                AsyncSnapshot<FfiListMember> snapshot,
+              ) {
+                if (snapshot.hasData) {
+                  return Text(
+                    '${snapshot.requireData.length.toString()} Members',
+                    style: GoogleFonts.montserrat(
+                      color: AppColors.primaryColor,
+                      fontSize: 12,
+                      fontWeight: FontWeight.w300,
+                    ),
+                  );
+                } else {
+                  return Container(
+                    height: 15,
+                    width: 15,
+                    child: CircularProgressIndicator(
+                      color: AppColors.primaryColor,
+                    ),
+                  );
+                }
+              },
+            ),
+          ],
+        ),
+        actions: [
+          SizedBox(
+            width: 70,
+          ),
+          Padding(
+            padding: const EdgeInsets.only(right: 10),
+            child: FutureBuilder<Uint8List>(
+              future: widget.room.avatar().then((fb) => fb.asTypedList()),
+              // a previously-obtained Future<String> or null
+              builder:
+                  (BuildContext context, AsyncSnapshot<List<int>> snapshot) {
+                if (snapshot.hasData) {
+                  return CircleAvatar(
+                    radius: 20,
+                    backgroundImage: MemoryImage(
+                      Uint8List.fromList(snapshot.requireData),
+                      scale: 0.5,
+                    ),
+                  );
+                } else {
+                  return CircleAvatar(
+                    radius: 20,
+                    backgroundColor: Colors.grey[600],
+                    child: SvgPicture.asset(
+                      'assets/images/people.svg',
+                      width: 20,
+                      height: 20,
+                    ),
+                  );
+                }
+              },
+            ),
+          ),
+        ],
+      ),
+      body: SafeArea(
+        bottom: false,
+        child: isLoading
+            ? Center(
+                child: Container(
+                  height: 15,
+                  width: 15,
+                  child: CircularProgressIndicator(
+                    color: AppColors.primaryColor,
+                  ),
+                ),
+              )
+            : Chat(
+                messages: _messages,
+                onSendPressed: _handleSendPressed,
+                user: _user,
+                //custom avatar builder
+                avatarBuilder: (userId) {
+                  return Padding(
+                    padding: const EdgeInsets.only(right: 10),
+                    child: CircleAvatar(
+                      radius: 15,
+                      child: Text(getNameFromId(userId)[0].toUpperCase()),
+                    ),
+                  );
```

Comment:
We should reuse the (then new) avatar-component here. The user-name-to-icon should only be a fallback.